driver-core/src/main/java/com/datastax/driver/core/JdkSSLOptions.java

```
@@ -52,7 +53,12 @@ protected JdkSSLOptions(SSLContext context, String[] cipherSuites) {
 
     @Override
     public SslHandler newSSLHandler(SocketChannel channel) {
-        SSLEngine engine = newSSLEngine(channel);
+        return newSSLHandler(channel, null);
```

Now that the class implements `RemoteEndpointAwareSSLOptions`, this method should never get called by our code. We could throw an `UnsupportedOperationException` or `AssertionError` here to ensure that this is the case.

---

Agreed.

However your remark made me think of the following situation: If someone is extending `JdkSSLOptions` and overriding this method to customize the `SslHandler`, he will experience runtime problems because his overridden method will not be called anymore. Granted, users should override `newSSLEngine` rather than `newSSLHandler`, but it is not forbidden to do so. Maybe something worth mentioning in the upgrade guide?

---

I'm even wondering if this wouldn't be a showstopper actually. I think it would be safer to create subclasses, i.e. `RemoteEndpointAwareJdkSSLOptions extends JdkSSLOptions`.